mobile/library/common/config/config.cc

```diff
@@ -538,5 +540,44 @@ stats_sinks: *stats_sinks
 R"(
         overload:
           global_downstream_max_connections: 0xffffffff # uint32 max
+#{custom_layers}
 )";
 // clang-format on
+
+const char* xds_cluster_insert = R"(
+    - name: some_static_layer
+      static_layer:
+        envoy.reloadable_features.test_feature_false: True
+    - name: some_rtds_layer
+      rtds_layer:
+        name: some_rtds_layer
+        rtds_config:
+          initial_fetch_timeout:
+            seconds: 1
+          resource_api_version: V3
+          api_config_source:
+            api_type: {}
+            transport_api_version: V3
+            grpc_services:
+              envoy_grpc:
```

Comment:
While Envoy Mobile has google_grpc disabled by default, we will need to enable it for the Traffic Director testing amongst other use cases (some users may want to enable google_grpc for the same reasons).  So we probably also want to make this configurable through the APIs, or provide a separate API, e.g. addGoogleGrpcXdsCluster vs. addEnvoyGrpcXdsCluster.

Reply:
I'm inclined to get Lyft's opinion on this but as currently we only have one use case for xDS which uses google's gRPC, I'd be inclined to as we add API docs, add an API doc saying that addWhateverWeCallThis requires building with --build_flags_for_google_grpc.  when/if folks want to do xDS with non-google-gRPC we could add a boolean to the function indicating which client should be used.

Reply:
The solution proposed by Alyssa sounds good to me. It's fine to keep xds coupled with google grpc client as long as it's well documented behavior.

Reply:
That sounds fine to me as well, but then, we'd have to update the integration tests to use google_grpc instead

Reply:
Working on switching to google_grpc now